OA-PASS/metadata-schemas
# Add support for Harvard schemas

* Adds Harvard-specific `global`, `common`, `nihms`, and `dash` schemas, and associated test data
  * The harvard global schema adds new fields `publicationType` and `firstAuthorAffiliation`.  The dash schema uses these new fields
* Refactors the test data and associated tests to accommodate multiple institutions who have conflicting schemas (as jhu and harvard are)

Information for @jrmartino :
In the Harvard assets image, use URLs
* For NIHMS: `https://oa-pass.github.io/metadata-schemas/harvard/nihms.json`
* For DASH: `https://oa-pass.github.io/metadata-schemas/harvard/dash.json`
* Comon: `https://oa-pass.github.io/metadata-schemas/harvard/common.json`

## To Test
* Switch to this PR in git
* Do `docker-compose build`
* Go to `pass-docker`.  edit `docker-compose.yml` to set the image of `schemaservice` to `oapass/schema-service:latest`.  This will make it use the image you just built, from this PR's code
* Launch pass-docker via `docker-compose up -d`
* Create a file `patch.sparql` with the following contents:
```
PREFIX pass: <http://oapass.org/ns/pass#>

DELETE {?s pass:schemas ?o }
INSERT { <> pass:schemas <https://oa-pass.github.io/metadata-schemas/harvard/common.json> .
<> pass:schemas <https://oa-pass.github.io/metadata-schemas/harvard/dash.json>}
WHERE { ?s ?p ?o }
```
* Once Fedora is up, do `curl -v -u fedoraAdmin:moo -H "Content-Type: application/sparql-update" -X PATCH --data-binary "@patch.sparql" http://localhost:8080/fcrepo/rest/repositories/41/96/0a/92/41960a92-d3f8-4616-86a6-9e9cadc1a269`
* What you did was change the J10P repository to use the Harvard DASH schema instead of the j10p schema.  So now create a submission *WITH NO GRANTS* so that it will be deposited only in the j10p repository.  When you get to the form, look for the presence of DASH fields such as  "First author affiliation" and "publication type"
* Feel free to complete the submission and verify that you aren't presented with errors upon clicking the 'submit' button.  The deposit may or may not succeed, since Deposit Services will see fields it doesn't know about.
